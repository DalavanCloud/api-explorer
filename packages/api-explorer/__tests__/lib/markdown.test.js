const fs = require('fs');
const markdown = require('../../src/lib/markdown');

const fixture = fs.readFileSync(`${__dirname}/markdown.txt`, 'utf8');

test('should render markdown', () => {
  expect(markdown(fixture)).toMatchSnapshot();
});

test('should render empty string if nothing passed in', () => {
  expect(markdown('')).toBe('');
});

test('`correctnewlines` option', () => {
  expect(markdown('test\ntest\ntest', { correctnewlines: true })).toBe('<p>test\ntest\ntest</p>\n');
  expect(markdown('test\ntest\ntest', { correctnewlines: false })).toBe(
    '<p>test<br>test<br>test</p>\n',
  );
});

describe('`stripHtml` option', () => {
  test('should allow html by default', () => {
    expect(markdown('<p>Test</p>')).toBe('<p><p>Test</p></p>\n');
    expect(markdown('<p>Test</p>', { stripHtml: false })).toBe('<p><p>Test</p></p>\n');
  });

  test('should escape unknown tags', () => {
    expect(markdown('<unknown-tag>Test</unknown-tag>')).toBe(
      '<p>&lt;unknown-tag&gt;Test&lt;/unknown-tag&gt;</p>\n',
    );
  });

  test('should allow certain attributes', () => {
    expect(markdown('<p id="test">Test</p>')).toBe('<p><p id="test">Test</p></p>\n');
  });

  test('should strip unknown attributes', () => {
    expect(markdown('<p unknown="test">Test</p>')).toBe('<p><p>Test</p></p>\n');
  });

  test('should escape everything if `stripHtml=true`', () => {
    expect(markdown('<p>Test</p>', { stripHtml: true })).toBe('<p>&lt;p&gt;Test&lt;/p&gt;</p>\n');
  });
});
