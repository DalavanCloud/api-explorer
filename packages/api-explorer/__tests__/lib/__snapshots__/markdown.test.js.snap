// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render markdown 1`] = `
"<p><img src=\\"http://example.com/image.png\\" alt=\\"Image\\"></p>
<ul>
<li >listitem1</li><li style=\\"list-style: none\\" class=\\"checklist\\"><input type=\\"checkbox\\" disabled> checklistitem1</li></ul>
<div class=\\"marked-table\\"><table>
<thead>
<tr>
<th>Tables</th>
<th style=\\"text-align:center\\">Are</th>
<th style=\\"text-align:right\\">Cool</th>
</tr>
</thead>
<tbody>
<tr>
<td>col 3 is</td>
<td style=\\"text-align:center\\">right-aligned</td>
<td style=\\"text-align:right\\">$1600</td>
</tr>
<tr>
<td>col 2 is</td>
<td style=\\"text-align:center\\">centered</td>
<td style=\\"text-align:right\\">$12</td>
</tr>
<tr>
<td>zebra stripes</td>
<td style=\\"text-align:center\\">are neat</td>
<td style=\\"text-align:right\\">$1</td>
</tr>
</tbody>
</table></div>
<h1 class=\\"header-scroll\\"><div class=\\"anchor waypoint\\" id=\\"section-h1\\"></div>h1<a class=\\"fa fa-anchor\\" href=\\"#section-h1\\"></a></h1>
<h2 class=\\"header-scroll\\"><div class=\\"anchor waypoint\\" id=\\"section-h2\\"></div>h2<a class=\\"fa fa-anchor\\" href=\\"#section-h2\\"></a></h2>
<h3 class=\\"header-scroll\\"><div class=\\"anchor waypoint\\" id=\\"section-h3\\"></div>h3<a class=\\"fa fa-anchor\\" href=\\"#section-h3\\"></a></h3>
<p><a href=\\"http://example.com\\" target=\\"_self\\">link</a><br><a href=\\"\\" target=\\"_self\\">xss</a><br><a href=\\"\\" ui-sref=\\"docs.show({'doc': 'slug'})\\" class=\\"doc-link\\" data-sidebar=\\"slug\\">doc</a><br><a href=\\"/reference#slug\\" target=\\"_self\\">ref</a><br><a href=\\"\\" ui-sref=\\"blog.show({'blog': 'slug'})\\">blog</a><br><a href=\\"page:slug\\" ui-sref=\\"custompages.show({'custompage': 'slug'})\\">page</a></p>
<p><img src=\\"/img/emojis/joy.png\\" alt=\\":joy+:\\" title=\\":joy:\\" class=\\"emoji\\" align=\\"absmiddle\\" height=\\"20\\" width=\\"20\\"><br><i class=\\"fa fa-lock\\"></i><br>:unknown-emoji:</p>
<pre><code class=\\"lang-js\\"><span class=\\"cm-s-neo\\">var a = 1;</span>
</code></pre>
"
`;
